content/sensu-core/2.0/reference/entities.md

```diff
@@ -4,82 +4,355 @@ description: "The entities reference guide."
 
-### What is an agent entity?
+## Entities specification
 
-Agent entities are monitoring agents, which are installed and run on every system that needs to be monitored. The entity is responsible for registering the system with the Sensu backend service, sending keepalive messages (the Sensu heartbeat mechanism), and executing monitoring checks. Each entity is a member of one or more **subscriptions** – a list of roles and/or responsibilities assigned to the agent entity (e.g. a webserver, database, etc). Sensu entities will "subscribe" to (or watch for) check requests published by the Sensu server (via the Sensu Transport), execute the corresponding requests locally, and publish the results of the check back to the transport (to be processed by a Sensu server).
+### Entity Attributes
 
-### Creating an agent entity
+ID           | 
+-------------|------ 
+description  | The unique ID of the entity, validated with go regex [`\A[\w\.\-]+\z`](https://regex101.com/r/zo9mQU/2)
+required     | true
+type         | string 
+example      | {{< highlight shell >}}"ID": "example-hostname"{{</ highlight >}}
 
-To create an agent entity, `sensu-agent` must be installed. Upon spinning up the sensu-agent, you can use several flags to customize the properties of the entity. In the example below, the optional flags are used to describe the default values of the agent entity that gets created upon starting sensu-agent. Other flags/properties you can set for your agent entity include: `--deregister`, `--subscriptions`, `--deregistration-handler`, and `--redact`.
+class        | 
+-------------|------ 
+description  | The entity type, validated with go regex [`\A[\w\.\-]+\z`](https://regex101.com/r/zo9mQU/2)
+required     | true
+type         | string 
+example      | {{< highlight shell >}}"class": "agent"{{</ highlight >}}
 
-{{< highlight shell >}}
-sensu-agent start --id hostname --environment default --organization default --keepalive-timeout 120 --user agent
-{{< /highlight >}}
+subscriptions| 
+-------------|------ 
+description  | A list of subscription names for the entity. 
+required     | false 
+type         | array 
+default      | The entity-specific subscription.
```

Comment:
Could we give an example or the format specs, i.e., `entity:{ID}`?

Reply:
absolutely.